src/InertialSense.cpp

```diff
@@ -947,45 +930,61 @@ void InertialSense::ProcessRxData(port_handle_t port, p_data_t* data)
         return;
     }
 
-    ISDevice* device = getDevice(port);
-    if (!device) {
-        if (data->hdr.id != DID_DEV_INFO)
-            return;
-    }
-
-    switch (data->hdr.id)
-    {
-        case DID_DEV_INFO:
-            if (!device) {
-                device = deviceManager.registerNewDevice(port, *(dev_info_t*)data->ptr);
-                device->devInfo.hdwRunState = HDW_STATE_APP; // we know this for a fact
-            } else {
-                device->devInfo = *(dev_info_t*)data->ptr;
-            }
-            break;
-        case DID_SYS_CMD:
-            device->sysCmd = *(system_command_t*)data->ptr;
-            break;
-        case DID_SYS_PARAMS:
-            copyDataPToStructP(&(device->sysParams), data, sizeof(sys_params_t));
-            DEBUG_PRINT("Received DID_SYS_PARAMS\n");
-            break;
-        case DID_FLASH_CONFIG:
-            copyDataPToStructP(&(device->flashCfg), data, sizeof(nvm_flash_cfg_t));
-            if (dataOverlap(offsetof(nvm_flash_cfg_t, checksum), 4, data))
-            {   // Checksum received
-                device->sysParams.flashCfgChecksum = device->flashCfg.checksum;
-            }
-            DEBUG_PRINT("Received DID_FLASH_CONFIG\n");
-            break;
-        case DID_FIRMWARE_UPDATE:
-            // we don't respond to messages if we don't already have an active Updater
-            if (device->fwUpdater) {
-                device->fwUpdater->fwUpdate_processMessage(data->ptr, data->hdr.size);
-                device->fwUpdate();
-            }
-            break;
-    }
+    // THIS IS MOVED TO ISDevice::ProcessRxData()
```

Comment:
If there is no need to keep this code, we should just remove it.  Git can serve as our history book.